includes/wc-user-functions.php

```diff
@@ -63,7 +63,7 @@ function wc_create_new_customer( $email, $username = '', $password = '', $args =
 
 		// Handle password creation.
 		$password_generated = false;
-		if ( 'yes' === get_option( 'woocommerce_registration_generate_password' ) && empty( $password ) ) {
+		if ( 'yes' === get_option( 'woocommerce_registration_generate_password' ) && empty( $password ) && ( ! isset( $args[ 'user_creation' ] ) || 'no' === get_option('woocommerce_registration_generate_password_allow_overwrite')) ) {
```

Comment:
Let me quickly unwrap this into sentences:

Only create a password if
1. Generate password is on
2. The password field is empty
3. Either
  a. user_creation is not set
  b. customers cannot supply a custom password and overwrite the autogenerated password.

I do not fully understand why 3. was added in the first place. Can you explain the intentions?
This means that if you activate the overwrite feature, no password will me generated automatically?

Reply:
My intention was to require the password on the my-account page (where you can create a user).
Any time woocommerce_registration_generate_password is on, the password field is optional.

Therefore I wanted to check whether the customer is on the my-account page and wants to create a new account without password or whether the account is created via the checkout process.